core/src/main/scala/org/genivi/sota/core/db/Campaigns.scala

```diff
@@ -0,0 +1,199 @@
+
+  // scalastyle:off
+  class CampaignTable(tag: Tag) extends Table[CampaignMeta](tag, "Campaign") {
+    def id = column[Id]("uuid")
+    def namespace = column[Namespace]("namespace")
+    def name = column[String]("name")
+    def launched = column[Boolean]("launched")
+    def packageName = column[PackageId.Name]("package_name")
+    def packageVersion = column[PackageId.Version]("package_version")
+
+    def pk = primaryKey("pk_campaign", id)
+
+    private def mPackageId(name0: Option[PackageId.Name],
+                           version0: Option[PackageId.Version]): Option[PackageId] = for
+      { name <- name0
+        version <- version0
+      } yield PackageId(name, version)
+
+    def * = (id, namespace, name, launched, packageName.?, packageVersion.?).shaped <>
+      ({case (id,ns, name, launched, pkgName, pkgVer) =>
+        CampaignMeta(id,ns, name, launched, mPackageId(pkgName, pkgVer))},
+       (x: CampaignMeta) => Some ((x.id, x.namespace, x.name, x.launched,
+                                   x.packageId.map(_.name), x.packageId.map(_.version))))
+  }
+
+  class CampaignGroups(tag: Tag) extends Table[(Campaign.Id, GroupInfo.Name)](tag, "CampaignGroups") {
+    def campaignId = column[Campaign.Id]("campaign_id")
+    def group = column[GroupInfo.Name]("group_name")
+
+    def pk = primaryKey("pk_campaign_group", (campaignId, group))
+
+    def * = (campaignId, group)
+
+  }
+  // scalastyle:on
+
+  val campaignsMeta = TableQuery[CampaignTable]
+  val campaignsGroups = TableQuery[CampaignGroups]
+
+  def canEdit(ns: Namespace, id: Id)
+             (implicit ec: ExecutionContext): DBIO[CampaignMeta] =
+    exists(ns, id).flatMap { camp =>
+      if (camp.launched){ DBIO.failed(SotaCoreErrors.CampaignLaunched) }
+       else {DBIO.successful(camp) }
+    }
+
+  def create(ns: Namespace, name: String)
+            (implicit ec: ExecutionContext): DBIO[Campaign.Id] = {
+    val id: Id = Id(refineV[ValidId](UUID.randomUUID.toString).right.get)
+
+    val dbIO = for {
+      _ <- exists(id).asTry.flatMap {
+        case Success(_) => DBIO.failed(SotaCoreErrors.ConflictingCampaign)
+        case Failure(_) => DBIO.successful(())
+      }
+      _ <- notConflicts(ns, name)
+      _ <- campaignsMeta += CampaignMeta(id, ns, name)
+    } yield id
+
+    dbIO.transactionally
+  }
+
+  def delete(ns: Namespace, id: Id)
+            (implicit ec: ExecutionContext): DBIO[Unit] = {
+    val dbIO = for {
+      _ <- exists(ns, id)
+      _ <- campaignsMeta.filter(c => c.namespace === ns && c.id === id).delete
+      _ <- campaignsGroups.filter(_.campaignId === id).delete
+    } yield ()
+
+    dbIO.transactionally
+  }
+
+  def exists(id: Id)
+            (implicit ec: ExecutionContext): DBIO[CampaignMeta] =
+    campaignsMeta
+      .filter(_.id === id)
+      .result
+      .headOption
+      .flatMap(_.fold[DBIO[CampaignMeta]](DBIO.failed(SotaCoreErrors.MissingCampaign))(DBIO.successful))
+
+
+  def exists(ns: Namespace, id: Id)
```

Comment:
namespace will not be here anymore, but anyway `id` is PK so you don't need to filter more than that.